Providers/Scripts/2.4x-2.5x/Scripts/nxOMSPlugin.py

```diff
@@ -312,6 +361,10 @@ def check_all_files(src, dest):
             if os.path.isfile(full_dest_file):
                 if CompareFiles(full_dest_file, full_src_file, 'md5') == -1:
                     return False
+                if is_exec:
+                    mode = os.stat(full_dest_file).st_mode
+                    if mode & 0x555 != 0x555:
```

Comment:
Please test this carefully; I haven't worked too much with binary operations in Ruby before.